graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1012,6 +1015,98 @@ def create_node_id_map_from_nodes_df(self, node_df: DataFrame, node_col: str) ->
+                    parent_path = ""
+                filename_pattern = os.path.basename(file_path)
+
+                # Get bucket and prefix for S3
+                input_bucket, input_prefix = s3_utils.extract_bucket_and_key(self.input_prefix)
+
+                # Combine parent path with input prefix if needed
+                if parent_path:
+                    if input_prefix:
+                        full_prefix = f"{input_prefix}/{parent_path}"
+                    else:
+                        full_prefix = parent_path
+                else:
+                    full_prefix = input_prefix
+
+                # List all objects under the prefix
+                all_objects = s3_utils.list_s3_objects(input_bucket, full_prefix)
+
+                # Filter objects that match the wildcard pattern
+                for obj_key in all_objects:
+                    obj_filename = os.path.basename(obj_key)
+                    # Only matches if the filename matches the pattern
+                    if fnmatch.fnmatch(obj_filename, filename_pattern):
+                        # Get relative path from input prefix
+                        if input_prefix:
+                            rel_path = (
+                                obj_key[len(input_prefix) + 1 :]
+                                if obj_key.startswith(f"{input_prefix}/")
+                                else obj_key
+                            )
+                        else:
+                            rel_path = obj_key
+                        expanded_files.append(rel_path)
+
+                if not expanded_files:
+                    logging.warning(
+                        "No files found matching pattern '%s' "
+                        "in S3 bucket '%s' under prefix '%s'",
+                        file_path,
+                        input_bucket,
+                        full_prefix,
+                    )
+            else:
+                # Handle local filesystem wildcards
+                full_pattern = os.path.join(self.input_prefix, file_path)
+                matched_files = glob.glob(full_pattern)
+
+                if not matched_files:
+                    logging.warning("No files found matching pattern '%s'", full_pattern)
+
+                # Convert absolute paths back to relative paths
+                for matched_file in matched_files:
+                    rel_path = os.path.relpath(matched_file, self.input_prefix)
+                    expanded_files.append(rel_path)
+
+        # Process each file path, expanding wildcards if needed
+        for file_path in files:
+            if "*" in file_path:
+                handle_wildcard(file_path, expanded_files)
+            else:
+                # No wildcard, use the file path as is
+                expanded_files.append(file_path)
+
+        # If no files were found after expansion, log a warning and fall back to original list
+        if not expanded_files and any("*" in f for f in files):
```

Comment:
Shall we expect an error here? I see it means there are no file input for this node type.